src/zarr/chunk_grids.py

```diff
@@ -22,13 +22,13 @@
 @dataclass(frozen=True)
 class ChunkGrid(Metadata):
     @classmethod
-    def from_dict(cls, data: dict[str, JSON]) -> ChunkGrid:
+    def from_dict(cls, data: dict[str, JSON] | ChunkGrid) -> ChunkGrid:
```

Comment:
your change is fine but I don't know why `from_dict` accepts a `ChunkGrid` object. 🤷